File: backend/app/schemas/project_task.py
from pydantic import BaseModel
from datetime import datetime


class ProjectTaskCreate(BaseModel):
    """POST リクエスト時に受け取るデータ"""
    project_id: int
    content: str


class ProjectTaskUpdate(BaseModel):
    """PATCH リクエスト時に受け取るデータ（完了/未完了の切り替え）"""
    is_done: bool


class ProjectTaskResponse(BaseModel):
    """レスポンスとして返すデータ"""
    id: int
    project_id: int
    content: str
    is_done: bool
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/projectTask.ts
// GET レスポンス用（FastAPI の ProjectTaskResponse に対応）
export type ProjectTask = {
  id: number;
  project_id: number;
  content: string;
  is_done: boolean;
  created_at: string;
};

// POST リクエスト用（FastAPI の ProjectTaskCreate に対応）
export type ProjectTaskCreate = {
  project_id: number;
  content: string;
};

// PATCH リクエスト用（FastAPI の ProjectTaskUpdate に対応）
export type ProjectTaskUpdate = {
  is_done: boolean;
};
